src/subprocess_unix.cpp

```diff
@@ -11,47 +11,30 @@
 
 UnixImpl::UnixImpl():
   m_pid(0),
-  m_running(false),
-  m_mutex(PTHREAD_MUTEX_INITIALIZER),
-  m_waitingThread()
+  m_running(false)
 {
 }
 
 UnixImpl::~UnixImpl()
 {
   kill();
 // Android has no pthread_cancel :(
-#ifdef __ANDROID__
-  pthread_kill(m_waitingThread, SIGUSR1);
-#else
-  pthread_cancel(m_waitingThread);
-#endif
-  pthread_join(m_waitingThread, nullptr);
+  m_waitingThread.join();
 }
 
-#ifdef __ANDROID__
-void thread_exit_handler(int sig) {
-  pthread_exit(0);
-}
-#endif
-
 void* UnixImpl::waitForPID(void* _self)
 {
-#ifdef __ANDROID__
-  struct sigaction actions;
-  memset(&actions, 0, sizeof(actions));
-  sigemptyset(&actions.sa_mask);
-  actions.sa_flags = 0;
-  actions.sa_handler = thread_exit_handler;
-  sigaction(SIGUSR1, &actions, NULL);
-#endif
-
   UnixImpl* self = static_cast<UnixImpl*>(_self);
-  waitpid(self->m_pid, NULL, 0);
+  while (true) {
+    if (!waitpid(self->m_pid, NULL, WNOHANG)) {
+      break;
+    }
+    if (self->m_shouldQuit) {
```

Comment:
I couldn't find where `m_shouldQuit` is set

Reply:
Added in the destructor.